lasagne/layers/base.py

```diff
@@ -39,49 +37,56 @@ def __init__(self, incoming, name=None):
         else:
             self.input_shape = incoming.output_shape
             self.input_layer = incoming
+
         self.name = name
+        self.params = OrderedDict()
 
     @property
     def output_shape(self):
         return self.get_output_shape_for(self.input_shape)
 
-    def get_params(self):
+    def get_params(self, **tags):
         """
         Returns a list of all the Theano variables that parameterize the layer.
 
+        By default, all parameters that were registered with
+        ``Layer.add_param()`` will be returned, in the order they were
+        registered. The list can optionally be filtered by specifying tags
+        as keyword arguments.
```

Comment:
I'm wondering if this should be targetted more towards users of layers, who don't know about `add_param`. An alternative might be:
```
By default, all parameters that participate in the forward pass will be returned (in the
order they were registered in the Layer's constructor via :meth:`add_param()`). The list
can optionally be filtered by specifying tags as keyword arguments. For example,
``trainable=True`` will only return trainable parameters, and ``regularizable=True``
will only return parameters that can be regularized (e.g, by L2 decay).
```
Feel free to mix and match if you agree.